libsmtutil/SolverInterface.h

```diff
@@ -262,23 +262,23 @@ class Expression
 	}
 	friend Expression operator+(Expression _a, Expression _b)
 	{
-		return Expression("+", std::move(_a), std::move(_b), Kind::Int);
+		return Expression("+", {std::move(_a), std::move(_b)}, SortProvider::sintSort);
```

Comment:
@chriseth actually these should use `_a`'s sort, as you pointed out at some point